controllerstatus/controllerstatus.go

```diff
@@ -0,0 +1,79 @@
+package controllerstatus
+
+import (
+	"sync"
+	"time"
+
+	"k8s.io/klog"
+)
+
+// Interface an interface for controller status
+type Interface interface {
+	CurrentStatus() (summary Summary, ready bool)
+}
+
+// Operation describing the operation
+type Operation string
+
+const (
+	// DownloadingReport specific flag for Smart Proxy report downloading process.
+	DownloadingReport Operation = "DownloadingReport"
```

Comment:
I think we can remove this one. This is for smart proxy and we won't be probably downloading anything :)

Reply:
Well now when I am looking it again, I guess we don't need this status controller. It's for setting and controlling the operator status.

Reply:
I removed the `DownloadingReport` operation. I think we will want some mechanism to pass status back to the operator so it can update its status. Let's keep this for the meantime until we determine how this information will be reflected.

Reply:
Yeah good point. My thinking was we would return only HTTP code back to an operator and let it update its status accordingly, but I don't have any strong preference right now :-).

Reply:
I think status is good. Also would be good to return the `x-rh-insights-request-id`.